Re-enable builds on OSX64 when it no longer segfaults on `import deepspeed`

### Solution to issue cannot be found in the documentation.

- [X] I checked the documentation.

### Issue

The segfault was first seen in #38, and described in detail at https://github.com/conda-forge/deepspeed-feedstock/pull/40#issuecomment-1817976141. Opening this issue as a reminder to re-enable the OSX64 builds at some point.

**Note**: OSX-ARM64 builds (i.e. the Apple Silicon builds) is still ok, at least as mentioned in https://github.com/conda-forge/deepspeed-feedstock/pull/42#issuecomment-1858768696. Support for these osx-arm64 chips was added in https://github.com/microsoft/DeepSpeed/pull/3907.

Traceback error from https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=845204&view=logs&j=4b3757f9-99c5-5891-6c0c-93d7030cdf7d&t=f047df50-a233-5ae9-bf8f-c7dd8e5684ad&l=4170:

```python-traceback
export PREFIX=/Users/runner/miniforge3/conda-bld/deepspeed_1702712167249/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_plac
export SRC_DIR=/Users/runner/miniforge3/conda-bld/deepspeed_1702712167249/test_tmp
import: 'deepspeed'
2023-12-16 07:43:06.688 python[6085:57815] -[MPSGraphExecutionDescriptor setEnableCommitAndContinue:]: unrecognized selector sent to instance 0x7f8fa686c870
2023-12-16 07:43:06.700 python[6085:57815] *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '-[MPSGraphExecutionDescriptor setEnableCommitAndContinue:]: unrecognized selector sent to instance 0x7f8fa686c870'
*** First throw call stack:
(
	0   CoreFoundation                      0x00007fff205e788f __exceptionPreprocess + 242
	1   libobjc.A.dylib                     0x00007fff20320bbb objc_exception_throw + 48
	2   CoreFoundation                      0x00007fff2066a3d8 -[NSObject(NSObject) __retain_OA] + 0
	3   CoreFoundation                      0x00007fff2054f99d ___forwarding___ + 1448
	4   CoreFoundation                      0x00007fff2054f368 _CF_forwarding_prep_0 + 120
	5   libtorch_cpu.dylib                  0x0000000112a5f73d _ZN2at3mps9MPSStreamC2EN3c106StreamE + 189
	6   libtorch_cpu.dylib                  0x0000000112a602ef _ZN2at3mps19getDefaultMPSStreamEv + 47
	7   libtorch_cpu.dylib                  0x0000000112a4848a _ZN2at3mps13HeapAllocator20MPSHeapAllocatorImplC2Ev + 762
	8   libtorch_cpu.dylib                  0x0000000112a4736c _ZN2at3mps12MPSAllocatorC2Ej + 444
	9   libtorch_cpu.dylib                  0x0000000112a47146 _ZN2at3mps16getIMPSAllocatorEb + 182
	10  libtorch_cpu.dylib                  0x0000000112a5189b _ZNK2at3mps8MPSHooks25getCurrentAllocatedMemoryEv + 11
	11  libtorch_python.dylib               0x000000010ccab2d0 _ZN5torch3mpsL32MPSModule_currentAllocatedMemoryEP7_objectS2_ + 48
	12  python                              0x0000000100953c7b cfunction_vectorcall_NOARGS + 91
	13  python                              0x0000000100a445dd _PyEval_EvalFrameDefault + 226365
	14  python                              0x0000000100a0af46 _PyEval_Vector + 1622
	15  python                              0x0000000100a038fe builtin___build_class__ + 382
	16  python                              0x0000000100953d7d cfunction_vectorcall_FASTCALL_KEYWORDS + 93
	17  python                              0x0000000100a445dd _PyEval_EvalFrameDefault + 226365
	18  python                              0x0000000100a0af46 _PyEval_Vector + 1622
	19  python                              0x0000000100a0a879 PyEval_EvalCode + 249
	20  python                              0x0000000100a0577d builtin_exec + 381
	21  python                              0x0000000100953d7d cfunction_vectorcall_FASTCALL_KEYWORDS + 93
	22  python                              0x0000000100a4aec1 _PyEval_EvalFrameDefault + 253217
	23  python                              0x00000001008f00ea _PyFunction_Vectorcall + 2634
	24  python                              0x00000001008ed689 _PyObject_VectorcallTstate.757 + 57
	25  python                              0x00000001008f3746 object_vacall + 374
	26  python                              0x00000001008f3572 PyObject_CallMethodObjArgs + 226
	27  python                              0x0000000100a89ef4 PyImport_ImportModuleLevelObject + 3172
	28  python                              0x0000000100a03ff1 builtin___import__ + 289
	29  python                              0x0000000100953d7d cfunction_vectorcall_FASTCALL_KEYWORDS + 93
	30  python                              0x0000000100a4aec1 _PyEval_EvalFrameDefault + 253217
	31  python                              0x00000001008f00ea _PyFunction_Vectorcall + 2634
	32  python                              0x00000001008ed689 _PyObject_VectorcallTstate.757 + 57
	33  python                              0x00000001008f3746 object_vacall + 374
	34  python                              0x00000001008f3572 PyObject_CallMethodObjArgs + 226
	35  python                              0x0000000100a896e2 PyImport_ImportModuleLevelObject + 1106
	36  python                              0x0000000100a334a0 _PyEval_EvalFrameDefault + 156416
	37  python                              0x0000000100a0af46 _PyEval_Vector + 1622
	38  python                              0x0000000100a0a879 PyEval_EvalCode + 249
	39  python                              0x0000000100a0577d builtin_exec + 381
	40  python                              0x0000000100953d7d cfunction_vectorcall_FASTCALL_KEYWORDS + 93
	41  python                              0x0000000100a4aec1 _PyEval_EvalFrameDefault + 253217
	42  python                              0x00000001008f00ea _PyFunction_Vectorcall + 2634
	43  python                              0x00000001008ed689 _PyObject_VectorcallTstate.757 + 57
	44  python                              0x00000001008f3746 object_vacall + 374
	45  python                              0x00000001008f3572 PyObject_CallMethodObjArgs + 226
	46  python                              0x0000000100a89ef4 PyImport_ImportModuleLevelObject + 3172
	47  python                              0x0000000100a334a0 _PyEval_EvalFrameDefault + 156416
	48  python                              0x0000000100a0af46 _PyEval_Vector + 1622
	49  python                              0x0000000100a0a879 PyEval_EvalCode + 249
	50  python                              0x0000000100a0577d builtin_exec + 381
	51  python                              0x0000000100953d7d cfunction_vectorcall_FASTCALL_KEYWORDS + 93
	52  python                              0x0000000100a4aec1 _PyEval_EvalFrameDefault + 253217
	53  python                              0x00000001008f00ea _PyFunction_Vectorcall + 2634
	54  python                              0x00000001008ed689 _PyObject_VectorcallTstate.757 + 57
	55  python                              0x00000001008f3746 object_vacall + 374
	56  python                              0x00000001008f3572 PyObject_CallMethodObjArgs + 226
	57  python                              0x0000000100a89ef4 PyImport_ImportModuleLevelObject + 3172
	58  python                              0x0000000100a03ff1 builtin___import__ + 289
	59  python                              0x0000000100953d7d cfunction_vectorcall_FASTCALL_KEYWORDS + 93
	60  python                              0x0000000100a4aec1 _PyEval_EvalFrameDefault + 253217
	61  python                              0x00000001008f00ea _PyFunction_Vectorcall + 2634
	62  python                              0x00000001008ed689 _PyObject_VectorcallTstate.757 + 57
	63  python                              0x00000001008f3746 object_vacall + 374
	64  python                              0x00000001008f3572 PyObject_CallMethodObjArgs + 226
	65  python                              0x0000000100a896e2 PyImport_ImportModuleLevelObject + 1106
	66  python                              0x0000000100a334a0 _PyEval_EvalFrameDefault + 156416
	67  python                              0x0000000100a0af46 _PyEval_Vector + 1622
	68  python                              0x0000000100a0a879 PyEval_EvalCode + 249
	69  python                              0x0000000100a0577d builtin_exec + 381
	70  python                              0x0000000100953d7d cfunction_vectorcall_FASTCALL_KEYWORDS + 93
	71  python                              0x0000000100a4aec1 _PyEval_EvalFrameDefault + 253217
	72  python                              0x00000001008f00ea _PyFunction_Vectorcall + 2634
	73  python                              0x00000001008ed689 _PyObject_VectorcallTstate.757 + 57
	74  python                              0x00000001008f3746 object_vacall + 374
	75  python                              0x00000001008f3572 PyObject_CallMethodObjArgs + 226
	76  python                              0x0000000100a89ef4 PyImport_ImportModuleLevelObject + 3172
	77  python                              0x0000000100a334a0 _PyEval_EvalFrameDefault + 156416
	78  python                              0x0000000100a0af46 _PyEval_Vector + 1622
	79  python                              0x0000000100a0a879 PyEval_EvalCode + 249
	80  python                              0x0000000100a0577d builtin_exec + 381
	81  python                              0x0000000100953d7d cfunction_vectorcall_FASTCALL_KEYWORDS + 93
	82  python                              0x0000000100a4aec1 _PyEval_EvalFrameDefault + 253217
	83  python                              0x00000001008f00ea _PyFunction_Vectorcall + 2634
	84  python                              0x00000001008ed689 _PyObject_VectorcallTstate.757 + 57
	85  python                              0x00000001008f3746 object_vacall + 374
	86  python                              0x00000001008f3572 PyObject_CallMethodObjArgs + 226
	87  python                              0x0000000100a89ef4 PyImport_ImportModuleLevelObject + 3172
	88  python                              0x0000000100a334a0 _PyEval_EvalFrameDefault + 156416
	89  python                              0x0000000100a0af46 _PyEval_Vector + 1622
	90  python                              0x0000000100a0a879 PyEval_EvalCode + 249
	91  python                              0x0000000100a0577d builtin_exec + 381
	92  python                              0x0000000100953d7d cfunction_vectorcall_FASTCALL_KEYWORDS + 93
	93  python                              0x0000000100a4aec1 _PyEval_EvalFrameDefault + 253217
	94  python                              0x00000001008f00ea _PyFunction_Vectorcall + 2634
	95  python                              0x00000001008ed689 _PyObject_VectorcallTstate.757 + 57
	96  python                              0x00000001008f3746 object_vacall + 374
	97  python                              0x00000001008f3572 PyObject_CallMethodObjArgs + 226
	98  python                              0x0000000100a89ef4 PyImport_ImportModuleLevelObject + 3172
	99  python                              0x0000000100a334a0 _PyEval_EvalFrameDefault + 156416
	100 python                              0x0000000100a0af46 _PyEval_Vector + 1622
	101 python                              0x0000000100a0a879 PyEval_EvalCode + 249
	102 python                              0x0000000100a0577d builtin_exec + 381
	103 python                              0x0000000100953d7d cfunction_vectorcall_FASTCALL_KEYWORDS + 93
	104 python                              0x0000000100a4aec1 _PyEval_EvalFrameDefault + 253217
	105 python                              0x00000001008f00ea _PyFunction_Vectorcall + 2634
	106 python                              0x00000001008ed689 _PyObject_VectorcallTstate.757 + 57
	107 python                              0x00000001008f3746 object_vacall + 374
	108 python                              0x00000001008f3572 PyObject_CallMethodObjArgs + 226
	109 python                              0x0000000100a89ef4 PyImport_ImportModuleLevelObject + 3172
	110 python                              0x0000000100a03ff1 builtin___import__ + 289
	111 python                              0x0000000100953d7d cfunction_vectorcall_FASTCALL_KEYWORDS + 93
	112 python                              0x0000000100a4aec1 _PyEval_EvalFrameDefault + 253217
	113 python                              0x00000001008f00ea _PyFunction_Vectorcall + 2634
	114 python                              0x00000001008ed689 _PyObject_VectorcallTstate.757 + 57
	115 python                              0x00000001008f3746 object_vacall + 374
	116 python                              0x00000001008f3572 PyObject_CallMethodObjArgs + 226
	117 python                              0x0000000100a896e2 PyImport_ImportModuleLevelObject + 1106
	118 python                              0x0000000100a334a0 _PyEval_EvalFrameDefault + 156416
	119 python                              0x0000000100a0af46 _PyEval_Vector + 1622
	120 python                              0x0000000100a0a879 PyEval_EvalCode + 249
	121 python                              0x0000000100a0577d builtin_exec + 381
	122 python                              0x0000000100953d7d cfunction_vectorcall_FASTCALL_KEYWORDS + 93
	123 python                              0x0000000100a4aec1 _PyEval_EvalFrameDefault + 253217
	124 python                              0x00000001008f00ea _PyFunction_Vectorcall + 2634
	125 python                              0x00000001008ed689 _PyObject_VectorcallTstate.757 + 57
	126 python                              0x00000001008f3746 object_vacall + 374
	127 python                              0x00000001008f3572 PyObject_CallMethodObjArgs + 226
	128 python                              0x0000000100a89ef4 PyImport_ImportModuleLevelObject + 3172
	129 python                              0x0000000100a03ff1 builtin___import__ + 289
	130 python                              0x0000000100953d7d cfunction_vectorcall_FASTCALL_KEYWORDS + 93
	131 python                              0x0000000100a4aec1 _PyEval_EvalFrameDefault + 253217
	132 python                              0x00000001008f00ea _PyFunction_Vectorcall + 2634
	133 python                              0x00000001008ed689 _PyObject_VectorcallTstate.757 + 57
	134 python                              0x00000001008f3746 object_vacall + 374
	135 python                              0x00000001008f3572 PyObject_CallMethodObjArgs + 226
	136 python                              0x0000000100a896e2 PyImport_ImportModuleLevelObject + 1106
	137 python                              0x0000000100a334a0 _PyEval_EvalFrameDefault + 156416
	138 python                              0x0000000100a0af46 _PyEval_Vector + 1622
	139 python                              0x0000000100a0a879 PyEval_EvalCode + 249
	140 python                              0x0000000100a0577d builtin_exec + 381
	141 python                              0x0000000100953d7d cfunction_vectorcall_FASTCALL_KEYWORDS + 93
	142 python                              0x0000000100a4aec1 _PyEval_EvalFrameDefault + 253217
	143 python                              0x00000001008f00ea _PyFunction_Vectorcall + 2634
	144 python                              0x00000001008ed689 _PyObject_VectorcallTstate.757 + 57
	145 python                              0x00000001008f3746 object_vacall + 374
	146 python                              0x00000001008f3572 PyObject_CallMethodObjArgs + 226
	147 python                              0x0000000100a89ef4 PyImport_ImportModuleLevelObject + 3172
	148 python                              0x0000000100a334a0 _PyEval_EvalFrameDefault + 156416
	149 python                              0x0000000100a0af46 _PyEval_Vector + 1622
	150 python                              0x0000000100a0a879 PyEval_EvalCode + 249
	151 python                              0x0000000100ab7e18 run_mod + 168
	152 python                              0x0000000100ab7c25 pyrun_file + 133
	153 python                              0x0000000100ab7703 _PyRun_SimpleFileObject + 275
	154 python                              0x0000000100ab709f _PyRun_AnyFileObject + 143
	155 python                              0x0000000100adb62b pymain_run_file_obj + 267
	156 python                              0x0000000100adb0a5 pymain_run_file + 85
	157 python                              0x0000000100adaa1e Py_RunMain + 1966
	158 python                              0x000000010087d5a8 main + 56
	159 libdyld.dylib                       0x00007fff20490f3d start + 1
)
libc++abi: terminating with uncaught exception of type NSException
/Users/runner/miniforge3/conda-bld/deepspeed_1702712167249/test_tmp/conda_test_runner.sh: line 3:  6085 Abort trap: 6           "/Users/runner/miniforge3/conda-bld/deepspeed_1702712167249/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_plac/bin/python" -s "/Users/runner/miniforge3/conda-bld/deepspeed_1702712167249/test_tmp/
WARNING: Tests failed for deepspeed-0.12.2-cpu_py311h0112a6b_2.conda - moving package to /Users/runner/miniforge3/conda-bld/broken
DEBUG:conda_index.index:found subdirs {'noarch', 'osx-64'}
DEBUG:conda_index.index.sqlitecache:CondaIndexCache channel_root=/Users/runner/miniforge3/conda-bld, subdir=noarch db_filename=/Users/runner/miniforge3/conda-bld/noarch/.cache/cache.db cache_is_brand_new=False
DEBUG:conda_index.index.sqlitecache:noarch listdir
DEBUG:conda_index.index.sqlitecache:noarch save fs state
DEBUG:conda_index.index:noarch find packages to extract
DEBUG:conda_index.index:noarch extract 0 packages
INFO:conda_index.index:noarch cached 0 B from 0 packages at 0 B/second
DEBUG:conda_index.index.sqlitecache:CondaIndexCache channel_root=/Users/runner/miniforge3/conda-bld, subdir=osx-64 db_filename=/Users/runner/miniforge3/conda-bld/osx-64/.cache/cache.db cache_is_brand_new=False
DEBUG:conda_index.index.sqlitecache:osx-64 listdir
DEBUG:conda_index.index.sqlitecache:osx-64 save fs state
DEBUG:conda_index.index:osx-64 find packages to extract
DEBUG:conda_index.index:osx-64 extract 0 packages
INFO:conda_index.index:osx-64 cached 0 B from 0 packages at 0 B/second
INFO:conda_index.index:Subdir: noarch Gathering repodata
DEBUG:conda_index.index.sqlitecache:CondaIndexCache channel_root=/Users/runner/miniforge3/conda-bld, subdir=noarch db_filename=/Users/runner/miniforge3/conda-bld/noarch/.cache/cache.db cache_is_brand_new=False
INFO:conda_index.index:noarch Writing pre-patch repodata
DEBUG:conda_index.index:_maybe_write /Users/runner/miniforge3/conda-bld/noarch/repodata_from_packages.json to /Users/runner/miniforge3/conda-bld/noarch/repodata_from_packages.json
INFO:conda_index.index:noarch Applying patch instructions
INFO:conda_index.index:noarch Writing patched repodata
DEBUG:conda_index.index:noarch write patched repodata
DEBUG:conda_index.index:_maybe_write /Users/runner/miniforge3/conda-bld/noarch/repodata.json to /Users/runner/miniforge3/conda-bld/noarch/repodata.json
INFO:conda_index.index:noarch Building current_repodata subset
DEBUG:conda_index.index:noarch build current_repodata
INFO:conda_index.index:noarch Writing current_repodata subset
DEBUG:conda_index.index:noarch write current_repodata
DEBUG:conda_index.index:_maybe_write /Users/runner/miniforge3/conda-bld/noarch/current_repodata.json to /Users/runner/miniforge3/conda-bld/noarch/current_repodata.json
INFO:conda_index.index:noarch Writing index HTML
DEBUG:conda_index.index:noarch write index.html
DEBUG:conda_index.index:_maybe_write /Users/runner/miniforge3/conda-bld/noarch/index.html to /Users/runner/miniforge3/conda-bld/noarch/index.html
DEBUG:conda_index.index:noarch finish
INFO:conda_index.index:Completed noarch
INFO:conda_index.index:Subdir: osx-64 Gathering repodata
DEBUG:conda_index.index.sqlitecache:CondaIndexCache channel_root=/Users/runner/miniforge3/conda-bld, subdir=osx-64 db_filename=/Users/runner/miniforge3/conda-bld/osx-64/.cache/cache.db cache_is_brand_new=False
INFO:conda_index.index:osx-64 Writing pre-patch repodata
DEBUG:conda_index.index:_maybe_write /Users/runner/miniforge3/conda-bld/osx-64/repodata_from_packages.json to /Users/runner/miniforge3/conda-bld/osx-64/repodata_from_packages.json
DEBUG:conda_index.index:_maybe_write /Users/runner/miniforge3/conda-bld/osx-64/repodata_from_packages.json.bz2 to /Users/runner/miniforge3/conda-bld/osx-64/repodata_from_packages.json.bz2
DEBUG:conda_index.index:_maybe_remove /Users/runner/miniforge3/conda-bld/osx-64/repodata_from_packages.json.zst from /Users/runner/miniforge3/conda-bld/osx-64/repodata_from_packages.json.zst
INFO:conda_index.index:osx-64 Applying patch instructions
INFO:conda_index.index:osx-64 Writing patched repodata
DEBUG:conda_index.index:osx-64 write patched repodata
DEBUG:conda_index.index:_maybe_write /Users/runner/miniforge3/conda-bld/osx-64/repodata.json to /Users/runner/miniforge3/conda-bld/osx-64/repodata.json
DEBUG:conda_index.index:_maybe_write /Users/runner/miniforge3/conda-bld/osx-64/repodata.json.bz2 to /Users/runner/miniforge3/conda-bld/osx-64/repodata.json.bz2
DEBUG:conda_index.index:_maybe_remove /Users/runner/miniforge3/conda-bld/osx-64/repodata.json.zst from /Users/runner/miniforge3/conda-bld/osx-64/repodata.json.zst
INFO:conda_index.index:osx-64 Building current_repodata subset
DEBUG:conda_index.index:osx-64 build current_repodata
INFO:conda_index.index:osx-64 Writing current_repodata subset
DEBUG:conda_index.index:osx-64 write current_repodata
DEBUG:conda_index.index:_maybe_write /Users/runner/miniforge3/conda-bld/osx-64/current_repodata.json to /Users/runner/miniforge3/conda-bld/osx-64/current_repodata.json
DEBUG:conda_index.index:_maybe_write /Users/runner/miniforge3/conda-bld/osx-64/current_repodata.json.bz2 to /Users/runner/miniforge3/conda-bld/osx-64/current_repodata.json.bz2
DEBUG:conda_index.index:_maybe_remove /Users/runner/miniforge3/conda-bld/osx-64/current_repodata.json.zst from /Users/runner/miniforge3/conda-bld/osx-64/current_repodata.json.zst
INFO:conda_index.index:osx-64 Writing index HTML
DEBUG:conda_index.index:osx-64 write index.html
DEBUG:conda_index.index:_maybe_write /Users/runner/miniforge3/conda-bld/osx-64/index.html to /Users/runner/miniforge3/conda-bld/osx-64/index.html
DEBUG:conda_index.index:osx-64 finish
INFO:conda_index.index:Completed osx-64
DEBUG:conda_index.index:found subdirs {'noarch', 'osx-64'}
INFO:conda_index.index:Channeldata subdir: noarch
DEBUG:conda_index.index:noarch read repodata
DEBUG:conda_index.index.sqlitecache:CondaIndexCache channel_root=/Users/runner/miniforge3/conda-bld, subdir=noarch db_filename=/Users/runner/miniforge3/conda-bld/noarch/.cache/cache.db cache_is_brand_new=False
DEBUG:conda_index.index:noarch channeldata finished
INFO:conda_index.index:Channeldata subdir: osx-64
DEBUG:conda_index.index:osx-64 read repodata
DEBUG:conda_index.index.sqlitecache:CondaIndexCache channel_root=/Users/runner/miniforge3/conda-bld, subdir=osx-64 db_filename=/Users/runner/miniforge3/conda-bld/osx-64/.cache/cache.db cache_is_brand_new=False
DEBUG:conda_index.index:osx-64 channeldata finished
DEBUG:conda_index.index:_maybe_write /Users/runner/miniforge3/conda-bld/index.html to /Users/runner/miniforge3/conda-bld/index.html
DEBUG:conda_index.index:write channeldata
DEBUG:conda_index.index:_maybe_write /Users/runner/miniforge3/conda-bld/channeldata.json to /Users/runner/miniforge3/conda-bld/channeldata.json
TESTS FAILED: deepspeed-0.12.2-cpu_py311h0112a6b_2.conda
##[error]Bash exited with code '1'.
##[section]Finishing: Run OSX build
```

### Installed packages

```shell
annotated-types:   0.6.0-pyhd8ed1ab_0              conda-forge
bzip2:             1.0.8-h10d778d_5                conda-forge
ca-certificates:   2023.11.17-h8857fd0_0           conda-forge
colorama:          0.4.6-pyhd8ed1ab_0              conda-forge
deepspeed:         0.12.2-cpu_py311h0112a6b_2      local      
filelock:          3.13.1-pyhd8ed1ab_0             conda-forge
fsspec:            2023.12.2-pyhca7485f_0          conda-forge
gmp:               6.3.0-h93d8f39_0                conda-forge
gmpy2:             2.1.2-py311hc5b4402_1           conda-forge
hjson-py:          3.1.0-pyhd8ed1ab_0              conda-forge
icu:               73.2-hf5e326d_0                 conda-forge
jinja2:            3.1.2-pyhd8ed1ab_1              conda-forge
libabseil:         20230802.1-cxx17_h048a20a_0     conda-forge
libblas:           3.9.0-20_osx64_openblas         conda-forge
libcblas:          3.9.0-20_osx64_openblas         conda-forge
libcxx:            16.0.6-hd57cbcb_0               conda-forge
libexpat:          2.5.0-hf0c8a7f_1                conda-forge
libffi:            3.4.2-h0d85af4_5                conda-forge
libgfortran:       5.0.0-13_2_0_h97931a8_1         conda-forge
libgfortran5:      13.2.0-h2873a65_1               conda-forge
libhwloc:          2.9.3-default_h24e0189_1009     conda-forge
libiconv:          1.17-hd75f5a5_2                 conda-forge
liblapack:         3.9.0-20_osx64_openblas         conda-forge
libopenblas:       0.3.25-openmp_hfef2a42_0        conda-forge
libprotobuf:       4.24.4-h0ee05dc_0               conda-forge
libsqlite:         3.44.2-h92b6c6a_0               conda-forge
libuv:             1.46.0-h0c2f820_0               conda-forge
libxml2:           2.11.6-hc0ae0f7_0               conda-forge
libzlib:           1.2.13-h8a1eda9_5               conda-forge
llvm-openmp:       17.0.6-hb6ac08f_0               conda-forge
markupsafe:        2.1.3-py311h2725bcf_1           conda-forge
mkl:               2022.2.1-h44ed08c_16952         conda-forge
mpc:               1.3.1-h81bd1dd_0                conda-forge
mpfr:              4.2.1-h0c69b56_0                conda-forge
mpmath:            1.3.0-pyhd8ed1ab_0              conda-forge
ncurses:           6.4-h93d8f39_2                  conda-forge
networkx:          3.2.1-pyhd8ed1ab_0              conda-forge
numpy:             1.26.2-py311h93c810c_0          conda-forge
openssl:           3.2.0-hd75f5a5_1                conda-forge
packaging:         23.2-pyhd8ed1ab_0               conda-forge
pip:               23.3.1-pyhd8ed1ab_0             conda-forge
psutil:            5.9.5-py311h2725bcf_1           conda-forge
py-cpuinfo:        9.0.0-pyhd8ed1ab_0              conda-forge
pydantic:          2.5.2-pyhd8ed1ab_0              conda-forge
pydantic-core:     2.14.5-py311h5e0f0e4_0          conda-forge
pynvml:            11.5.0-pyhd8ed1ab_0             conda-forge
python:            3.11.7-h9f0c242_0_cpython       conda-forge
python_abi:        3.11-4_cp311                    conda-forge
pytorch:           2.1.0-cpu_mkl_py311h5cbed28_100 conda-forge
readline:          8.2-h9e318b2_1                  conda-forge
setuptools:        68.2.2-pyhd8ed1ab_0             conda-forge
sleef:             3.5.1-h6db0672_2                conda-forge
sympy:             1.12-pypyh9d50eac_103           conda-forge
tbb:               2021.11.0-he51d815_0            conda-forge
tk:                8.6.13-h1abcd95_1               conda-forge
tqdm:              4.66.1-pyhd8ed1ab_0             conda-forge
typing-extensions: 4.9.0-hd8ed1ab_0                conda-forge
typing_extensions: 4.9.0-pyha770c72_0              conda-forge
tzdata:            2023c-h71feb2d_0                conda-forge
wheel:             0.42.0-pyhd8ed1ab_0             conda-forge
xz:                5.2.6-h775f41a_0                conda-forge
```


### Environment info

```shell
active environment : base
   active env location : /Users/runner/miniforge3
           shell level : 1
      user config file : /Users/runner/.condarc
populated config files : /Users/runner/miniforge3/.condarc
         conda version : 23.11.0
   conda-build version : 3.27.0
        python version : 3.10.12.final.0
                solver : libmamba (default)
      virtual packages : __archspec=1=ivybridge
                         __conda=23.11.0=0
                         __osx=11.7.10=0
                         __unix=0=0
      base environment : /Users/runner/miniforge3  (writable)
     conda av data dir : /Users/runner/miniforge3/etc/conda
 conda av metadata url : None
          channel URLs : https://conda.anaconda.org/conda-forge/osx-64
                         https://conda.anaconda.org/conda-forge/noarch
         package cache : /Users/runner/miniforge3/pkgs
                         /Users/runner/.conda/pkgs
      envs directories : /Users/runner/miniforge3/envs
                         /Users/runner/.conda/envs
              platform : osx-64
            user-agent : conda/23.11.0 requests/2.31.0 CPython/3.10.12 Darwin/20.6.0 OSX/11.7.10 solver/libmamba conda-libmamba-solver/23.12.0 libmambapy/1.5.5
               UID:GID : 501:20
            netrc file : None
          offline mode : False
```

--- a/.azure-pipelines/azure-pipelines-osx.yml
+++ b/.azure-pipelines/azure-pipelines-osx.yml
@@ -8,6 +8,18 @@ jobs:
     vmImage: macOS-12
   strategy:
     matrix:
+      osx_64_python3.10.____cpython:
+        CONFIG: osx_64_python3.10.____cpython
+        UPLOAD_PACKAGES: 'True'
+      osx_64_python3.11.____cpython:
+        CONFIG: osx_64_python3.11.____cpython
+        UPLOAD_PACKAGES: 'True'
+      osx_64_python3.12.____cpython:
+        CONFIG: osx_64_python3.12.____cpython
+        UPLOAD_PACKAGES: 'True'
+      osx_64_python3.9.____cpython:
+        CONFIG: osx_64_python3.9.____cpython
+        UPLOAD_PACKAGES: 'True'
       osx_arm64_python3.10.____cpython:
         CONFIG: osx_arm64_python3.10.____cpython
         UPLOAD_PACKAGES: 'True'

--- a/.ci_support/osx_64_python3.10.____cpython.yaml
+++ b/.ci_support/osx_64_python3.10.____cpython.yaml
@@ -1,0 +1,37 @@
+MACOSX_DEPLOYMENT_TARGET:
+- '10.13'
+MACOSX_SDK_VERSION:
+- '10.13'
+c_compiler:
+- clang
+c_compiler_version:
+- '17'
+c_stdlib:
+- macosx_deployment_target
+c_stdlib_version:
+- '10.13'
+channel_sources:
+- conda-forge
+channel_targets:
+- conda-forge main
+cuda_compiler:
+- None
+cuda_compiler_version:
+- None
+cxx_compiler:
+- clangxx
+cxx_compiler_version:
+- '17'
+macos_machine:
+- x86_64-apple-darwin13.4.0
+pin_run_as_build:
+  python:
+    min_pin: x.x
+    max_pin: x.x
+python:
+- 3.10.* *_cpython
+target_platform:
+- osx-64
+zip_keys:
+- - c_compiler_version
+  - cxx_compiler_version

--- a/.ci_support/osx_64_python3.11.____cpython.yaml
+++ b/.ci_support/osx_64_python3.11.____cpython.yaml
@@ -1,0 +1,37 @@
+MACOSX_DEPLOYMENT_TARGET:
+- '10.13'
+MACOSX_SDK_VERSION:
+- '10.13'
+c_compiler:
+- clang
+c_compiler_version:
+- '17'
+c_stdlib:
+- macosx_deployment_target
+c_stdlib_version:
+- '10.13'
+channel_sources:
+- conda-forge
+channel_targets:
+- conda-forge main
+cuda_compiler:
+- None
+cuda_compiler_version:
+- None
+cxx_compiler:
+- clangxx
+cxx_compiler_version:
+- '17'
+macos_machine:
+- x86_64-apple-darwin13.4.0
+pin_run_as_build:
+  python:
+    min_pin: x.x
+    max_pin: x.x
+python:
+- 3.11.* *_cpython
+target_platform:
+- osx-64
+zip_keys:
+- - c_compiler_version
+  - cxx_compiler_version

--- a/.ci_support/osx_64_python3.12.____cpython.yaml
+++ b/.ci_support/osx_64_python3.12.____cpython.yaml
@@ -1,0 +1,37 @@
+MACOSX_DEPLOYMENT_TARGET:
+- '10.13'
+MACOSX_SDK_VERSION:
+- '10.13'
+c_compiler:
+- clang
+c_compiler_version:
+- '17'
+c_stdlib:
+- macosx_deployment_target
+c_stdlib_version:
+- '10.13'
+channel_sources:
+- conda-forge
+channel_targets:
+- conda-forge main
+cuda_compiler:
+- None
+cuda_compiler_version:
+- None
+cxx_compiler:
+- clangxx
+cxx_compiler_version:
+- '17'
+macos_machine:
+- x86_64-apple-darwin13.4.0
+pin_run_as_build:
+  python:
+    min_pin: x.x
+    max_pin: x.x
+python:
+- 3.12.* *_cpython
+target_platform:
+- osx-64
+zip_keys:
+- - c_compiler_version
+  - cxx_compiler_version

--- a/.ci_support/osx_64_python3.9.____cpython.yaml
+++ b/.ci_support/osx_64_python3.9.____cpython.yaml
@@ -1,0 +1,37 @@
+MACOSX_DEPLOYMENT_TARGET:
+- '10.13'
+MACOSX_SDK_VERSION:
+- '10.13'
+c_compiler:
+- clang
+c_compiler_version:
+- '17'
+c_stdlib:
+- macosx_deployment_target
+c_stdlib_version:
+- '10.13'
+channel_sources:
+- conda-forge
+channel_targets:
+- conda-forge main
+cuda_compiler:
+- None
+cuda_compiler_version:
+- None
+cxx_compiler:
+- clangxx
+cxx_compiler_version:
+- '17'
+macos_machine:
+- x86_64-apple-darwin13.4.0
+pin_run_as_build:
+  python:
+    min_pin: x.x
+    max_pin: x.x
+python:
+- 3.9.* *_cpython
+target_platform:
+- osx-64
+zip_keys:
+- - c_compiler_version
+  - cxx_compiler_version

--- a/README.md
+++ b/README.md
@@ -113,6 +113,34 @@ Current build status
                 </a>
               </td>
             </tr><tr>
+              <td>osx_64_python3.10.____cpython</td>
+              <td>
+                <a href="https://dev.azure.com/conda-forge/feedstock-builds/_build/latest?definitionId=18449&branchName=main">
+                  <img src="https://dev.azure.com/conda-forge/feedstock-builds/_apis/build/status/deepspeed-feedstock?branchName=main&jobName=osx&configuration=osx%20osx_64_python3.10.____cpython" alt="variant">
+                </a>
+              </td>
+            </tr><tr>
+              <td>osx_64_python3.11.____cpython</td>
+              <td>
+                <a href="https://dev.azure.com/conda-forge/feedstock-builds/_build/latest?definitionId=18449&branchName=main">
+                  <img src="https://dev.azure.com/conda-forge/feedstock-builds/_apis/build/status/deepspeed-feedstock?branchName=main&jobName=osx&configuration=osx%20osx_64_python3.11.____cpython" alt="variant">
+                </a>
+              </td>
+            </tr><tr>
+              <td>osx_64_python3.12.____cpython</td>
+              <td>
+                <a href="https://dev.azure.com/conda-forge/feedstock-builds/_build/latest?definitionId=18449&branchName=main">
+                  <img src="https://dev.azure.com/conda-forge/feedstock-builds/_apis/build/status/deepspeed-feedstock?branchName=main&jobName=osx&configuration=osx%20osx_64_python3.12.____cpython" alt="variant">
+                </a>
+              </td>
+            </tr><tr>
+              <td>osx_64_python3.9.____cpython</td>
+              <td>
+                <a href="https://dev.azure.com/conda-forge/feedstock-builds/_build/latest?definitionId=18449&branchName=main">
+                  <img src="https://dev.azure.com/conda-forge/feedstock-builds/_apis/build/status/deepspeed-feedstock?branchName=main&jobName=osx&configuration=osx%20osx_64_python3.9.____cpython" alt="variant">
+                </a>
+              </td>
+            </tr><tr>
               <td>osx_arm64_python3.10.____cpython</td>
               <td>
                 <a href="https://dev.azure.com/conda-forge/feedstock-builds/_build/latest?definitionId=18449&branchName=main">

--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -72,7 +72,6 @@ requirements:
     - libcusparse-dev
     {% endif %}
   run:
-    - ninja
     - hjson-py
     # NOTE(hadim): coming at https://github.com/conda-forge/staged-recipes/pull/19098
     # - ninja-python

--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -52,13 +52,13 @@ requirements:
     - libcusparse-dev                                  # [build_platform != target_platform]
     {% endif %}
   host:
-    - python
-    - pip
     - git
-    - oneccl-devel                                     # [cuda_compiler_version not in (undefined, 'None')]
-    - py-cpuinfo
     # For the `async_io` op
-    - libaio  # [linux]
+    - libaio                                           # [linux]
+    - oneccl-devel                                     # [cuda_compiler_version not in (undefined, 'None')]
+    - pip
+    - py-cpuinfo
+    - python
     # Leaving two dependencies helps rerender correctly
     # The first gets filled in by the global pinnings
     # The second gets the processor type

--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -64,6 +64,7 @@ requirements:
     # The second gets the processor type
     - pytorch                                          # [cuda_compiler_version not in (undefined, 'None')]
     - pytorch =*={{ torch_proc_type }}*                # [cuda_compiler_version not in (undefined, 'None')]
+    - setuptools
     {% if cuda_major >= 12 %}
     - cuda-profiler-api
     - libcublas-dev

--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -1,6 +1,6 @@
 {% set name = "deepspeed" %}
 {% set version = "0.15.1" %}
-{% set build = 0 %}
+{% set build = 1 %}
 
 {% set torch_proc_type = "cuda" if cuda_compiler_version != "None" else "cpu" %}
 

--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -27,7 +27,7 @@ build:
   number: {{ build }}
   string: cuda{{ cuda_compiler_version | replace('.', '') }}py{{ CONDA_PY }}h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}  # [cuda_compiler_version != "None"]
   string: cpu_py{{ CONDA_PY }}h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}  # [cuda_compiler_version == "None"]
-  skip: true  # [win or (osx and x86_64)]
+  skip: true  # [win]
 
 requirements:
   # NOTE(hadim): pytorch is only needed during the build in reqs.build and reqs.host when the ops are being built which

